File: src/Controlador/ControladorEstudante.java
package Controlador;

import java.util.Scanner;

import Leitor.LeitorEstudante;
import Modelo.Estudante;


public class ControladorEstudante implements IControlador{
	
    public Estudante busca(String matricula){
		LeitorEstudante lEstudante = LeitorEstudante.obterInstancia();
		return lEstudante.busca(matricula);
	}

	public void ler(Scanner scan){
		LeitorEstudante leitor = LeitorEstudante.obterInstancia();
		while(scan.hasNext()){
			Estudante estudante = leitor.ler(scan);
			leitor.anexaHash(estudante);
		}
	}
}

File: cpp_version/Controlador/ControladorEstudante.cpp
#include "ControladorEstudante.h"

Estudante* ControladorEstudante :: busca(string codigo){
    LeitorEstudante* leitor= LeitorEstudante :: obterInstancia();
    return leitor->busca(codigo);
}

void ControladorEstudante :: ler(ifstream* scan){
    LeitorEstudante* leitor = LeitorEstudante::obterInstancia();
    
    string linha;
    getline(*scan, linha);
    while (getline(*scan, linha)) {
        Tokenizer tok(linha, ';');
        vector<string> vec = tok.remaining();
        for (int i = 0; i < vec.size(); i++){
            vec[i] = trim(vec[i]);
        }
        Estudante* estudante = leitor->ler(vec);
        leitor->anexaHash(estudante);
    }
}

vector<Estudante*> ControladorEstudante :: lista(){
    LeitorEstudante* leitor = LeitorEstudante::obterInstancia();
    return leitor->obterEstudantes();
}